### DISCUSSION TITLE
$attributes onlyStartsWith and exceptStartsWith

### DISCUSSION THREAD

**Discussion by tanthammar** [laravel/framework#38290] [Ideas] [answered]
Regarding Blade components Attribute Bag.

When developing headless Blade Components I miss two methods on the `ComponentAttributeBag`

```blade
{{ $attributes->onlyStartsWith([...]) }}
{{ $attributes->exceptStartsWith([...]) }}
```

Example use case
```blade
<div {{ $attributes->onlyStartsWith(['wire:model', 'x-model']) }} >
    <input 
        {{ $attributes->exceptStartsWith(['wire:model', 'x-model', 'x-data'])->merge(...) }} 
        @change="$dispatch(...)"
    />
</div>
```

Current solution is a bit to verbose
```blade
<div {{ $attributes->whereStartsWith('wire:model') }} {{ $attributes->whereStartsWith('x-model') }} > 
    <input 
        {{ $attributes->except('x-data')->whereDoesntStartWith('wire:model')->whereDoesntStartWith('x-model')->merge(...) }} 
        @change="$dispatch(...)"
    />
</div>
```

**Comment by devcircus** [comment]
Maybe filter() would work well for these use-cases.

**Comment by ryangjchandler** [comment, marked as answer]
@tanthammar Despite the `whereStartsWith` and `whereDoesntStartWith` DocBlocks saying it only accepts a string, you can actually pass an array to these methods as it's being passed to `Str::startsWith($haystack, $needles)` which accepts an array too.

**Reply by tanthammar** [reply]
@ryangjchandler Clever you ;) Thank you!
